File: src/main/java/com/atlaspharmacy/atlaspharmacy/pharmacy/domain/Pricelist.java
package com.atlaspharmacy.atlaspharmacy.pharmacy.domain;

import com.atlaspharmacy.atlaspharmacy.medication.domain.Medication;
import com.atlaspharmacy.atlaspharmacy.schedule.domain.valueobjects.Period;

import javax.persistence.*;

@Entity
@Table(name = "pricelists")
public class Pricelist {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.MERGE)
    private Medication medication;

    private Long price;

    @Embedded
    @AttributeOverrides({
            @AttributeOverride( name = "startTime", column = @Column(name = "startPeriod")),
            @AttributeOverride( name = "endTime", column = @Column(name = "endPeriod"))
    })
    private Period validPeriod;
    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.MERGE)
    private Pharmacy pharmacy;

    public Pricelist() {
    }

    public Pricelist(Long id, Medication medication, Long price, Period validPeriod, Pharmacy pharmacy) {
        this.id = id;
        this.medication = medication;
        this.price = price;
        this.validPeriod = validPeriod;
        this.pharmacy = pharmacy;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Medication getMedication() {
        return medication;
    }

    public void setMedication(Medication medication) {
        this.medication = medication;
    }

    public Long getPrice() {
        return price;
    }

    public void setPrice(Long price) {
        this.price = price;
    }

    public Period getPeriod() {
        return validPeriod;
    }

    public void setPeriod(Period validPeriod) {
        this.validPeriod = validPeriod;
    }

    public Pharmacy getPharmacy() {
        return pharmacy;
    }

    public void setPharmacy(Pharmacy pharmacy) {
        this.pharmacy = pharmacy;
    }

}

File: frontend/src/app/model/pharmacy/pricelist.ts
// private Long id;
// private Long price;
// private MedicationDTO medication;
// private PharmacyDTO pharmacy;
// private Period period;

import { Medication } from "../medications/medication";
import { Pharmacy } from "./pharmacy";
import {Period} from "./../appointment/period"
export class Pricelist{
    public id : Number;
    public price : Number;
    public medication : Medication;
    public pharmacy : Pharmacy;
    public startPeriod : Date;
    public endPeriod : Date;

    constructor(id:Number,medication:Medication,price:Number,startPeriod:Date,endPeriod:Date,pharmacy:Pharmacy) {
      this.id = id;
      this.medication = medication;
      this.pharmacy = pharmacy;
      this.price = price;
      this.startPeriod = startPeriod;
      this.endPeriod = endPeriod;

    }
}
